content/spaces/locking.textile

```diff
@@ -0,0 +1,215 @@
+
+A lock will be in one of the following states:
+
+- @pending@ := A member has requested a lock by calling "@acquire()@":#acquire.
+- @locked@ := The lock is confirmed to be held by the requesting member.
+- @unlocked@ := The lock is confirmed to not be locked by the requesting member, or has been "released":#release by a member previously holding the lock.
+
+The following lock state transitions may occur. Only transitions that result in a @locked@ or @unlocked@ status will emit a lock event that members can "@subscribe()@":#subscribe to.
+
+* None --> @pending@: a member calls "@acquire()@":#acquire to request a lock.
+* @pending@ --> @locked@: the requesting member holds the lock.
+* @pending@ --> @unlocked@: the requesting member does not hold the lock, since another member already holds it.
+* @locked@ --> @unlocked@: the lock was either explicitly "released":#release by the member, or their request was invalidated by a concurrent request which took precedence.
+* @unlocked@ --> @locked@: the requesting member reacquired a lock they previously held.
+
+h2(#acquire). Acquire a lock
+
+Use the @acquire()@ method to attempt to acquire a lock with a given unique ID. Additional @attributes@ may be passed when trying to acquire a lock that can contain a set of arbitrary key-value pairs. An example of using @attributes@ is to store the component ID the lock relates to so that it can be easily updated in the UI with a visual indication of its lock status.
+
+A member must have been "entered":/spaces/space#enter into the space to acquire a lock.
+
+The following is an example of attempting to acquire a lock:
+
+```[javascript]
+const acquireLock = await space.locks.acquire(id);
+```
+
+The following is an example of passing a set of @attributes@ when trying to acquire a lock:
+
+```[javascript]
+const lockAttributes = new Map();
+lockAttributes.set('component', 'cell-d3');
+const acquireLock = await space.locks.acquire(id, { lockAttributes });
+```
+
+Once a member requests a lock by calling @acquire()@, the lock is temporarily in the "pending state":#states. An event will be emitted based on whether the lock request was successful (a status of @locked@) or invalidated (a status of @unlocked@).
+
+h2(#release). Release a lock
+
+Use the @release()@ method to explicitly release a lock once a member has finished editing the related component.
+
+The following is an example of releasing a lock:
+
+```[javascript]
+await space.locks.release(id);
+```
+
+Releasing a lock will emit a lock event with a "lock status":#states of @unlocked@.
+
+<aside data-type='note'>
+<p>When a member "leaves":/spaces/space#leave a space, their locks are automatically released.</p>
+</aside>
+
+h2(#subscribe). Subscribe to lock events
+
+Subscribe to lock events by registering a listener. Lock events are emitted whenever the "lock state":#states transitions. Use the @subscribe()@ method on the @locks@ namespace of the space to receive updates.
```

Comment:
emitted whenever the "lock state":#states transitions into either @locked@ or @unlocked@?